tasks/vmaas_sync/package_sync.go

```diff
@@ -218,15 +235,62 @@ func storePackageDetailsFrmPkgListItems(tx *gorm.DB, nameIDs map[string]int, pkg
 	}
 	utils.Log("packages", len(toStore)).Info("Collected packages to store")
 
+	err := toStore.storeOrUpdate(tx)
+	return err
+}
+
+//nolint:funlen
+func (pkgs pkgSlice) storeOrUpdate(tx *gorm.DB) error {
+	var toUpdate pkgSlice
+
+	nameIDEvra := make([][]interface{}, 0, len(pkgs))
+	toStore := make(pkgSlice, 0, len(pkgs))
+	for _, pkg := range pkgs {
+		nameIDEvra = append(nameIDEvra, []interface{}{pkg.NameID, pkg.EVRA})
+	}
+
+	tx.Where("(name_id, evra) IN ?", nameIDEvra).Find(&toUpdate)
+	for _, pkg := range pkgs {
```

Comment:
Faster than two loops inside would be create a map of `[name_id, evra]` and then single loop with checking presence in the map.

Reply:
```
		nameIDEVRAs = append(nameIDEVRAs, nameIDandEvra{pkg.NameID, pkg.EVRA})
```

Reply:
done